### PR TITLE
Python production virtualenv

### PR DESCRIPTION
**Pull request by bazsi**
This is the 9th and final piece of the original #4165 PR that implements support for an optional, production-time virtualenv. This is the piece that contains the NEWS file for the entire series.

### CODE REVIEW

**Comment by bazsi** [comment]
rebased against master.

File: news/feature-4165.md [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+Python virtualenv support for development use: syslog-ng is now capable of
+using a build-time virtualenv, where all Python development tools are
+automatically deployed by the build system.  You can control if you want to
+use this using the --with-python-packages configure parameter, the `venv`
+value denoting that you want to use the virtualenv, the `system` value
+meaning that you want to rely on system packages (as before).  Please note
+that syslog-ng has acquired quite a number of these development time
+dependencies with the growing number of functionality the Python binding
+offers, so using the `system` setting is considered advanced usage, meant to
+be used for distro packaging.
```
Comment (alltilla) [inline news/feature-4165.md:66]: As this is a breaking change regarding package maintainers, I think we should add a `packaging` news entry to this also.

File: CMakeLists.txt [hunk 2]
```diff
@@ -53,6 +53,8 @@ set(SYSLOG_NG_PACKAGE_NAME "${CMAKE_PROJECT_NAME}")
 set(SYSLOG_NG_PATH_XSDDIR "\${datadir}/syslog-ng/xsd")
 set(SYSLOG_NG_JAVA_MODULE_PATH "\${exec_prefix}/lib/syslog-ng/java-modules")
 set(SYSLOG_NG_PYTHON_MODULE_DIR "\${exec_prefix}/lib/syslog-ng/python")
+set(SYSLOG_NG_PYTHON_SYSCONF_MODULE_DIR "\${prefix}/etc/python")
+set(SYSLOG_NG_PYTHON_VENV_DIR "\${localstatedir}/python-venv")
```
Comment (alltilla) [inline CMakeLists.txt:57]: I think this should be `syslog-ng` specific, as the `${localstatedir}` (`/var`, `/usr/local/var`) can contain files and dirs from other projects/packages. If you agree, kindly update it here, at `configure.ac` and the newsfile entry, too. Thanks!

**Review by alltilla — COMMENTED** [review]
Nice feature, thanks!

I only had 2 minor comments, kindly consider them. Other than those, approve from me.

File: CMakeLists.txt (hunk 2 above)
Comment (bazsi) [reply CMakeLists.txt:57]: This is syslog-ng specific, normally, that's why it is `@localstatedir@` and not stateside

If you look at our packaging this is pointing to /var/lib/syslog-ng

File: news/feature-4165.md (hunk 1 above)
Comment (bazsi) [reply news/feature-4165.md:66]: I did that already in the version bump PR when I finalized the NEWS file there. Let me know if you find the wording unsatisfactory.

File: CMakeLists.txt (hunk 2 above)
Comment (alltilla) [reply CMakeLists.txt:57]: Thanks, I missed that.

File: news/feature-4165.md (hunk 1 above)
Comment (alltilla) [reply news/feature-4165.md:66]: It is perfect, thank you :)

**Review by alltilla — APPROVED** [review]

**Pull request merged**